Mage.Server.Plugins/Mage.Player.Human/src/mage/player/human/HumanPlayer.java

```diff
@@ -1398,6 +1397,40 @@ && lookAtFaceDownCard((Card) object, game, useableAbilities.size())) {
         return false;
     }
 
+    private boolean skipButtonActivated() {
+        return passedAllTurns || passedUntilEndStepBeforeMyTurn || passedTurn || passedUntilEndOfTurn
+                || passedUntilNextMain;
+    }
+
+    private boolean quickStopForLands(Game game) {
+        if (skipButtonActivated()) {
+            Battlefield battlefield = game.getBattlefield();
+            for (Permanent land : battlefield.getAllActivePermanents(filterLand, playerId, game)) {
+                if (land.canUseActivatedAbilities(game)) {
+                    String name = land.getName();
+                    if (name.equals("Maze of Ith") || name.equals("Mystifying Maze") || name.equals("Spires of Orazca")) {
+                        // Find valid targets
+                        for (Permanent attacker : battlefield.getAllActivePermanents(filterAttack, game)) {
+                            if (attacker.canBeTargetedBy(land, playerId, game)) {
```

Comment:
It's a declare blocker step -- there are already attacked permanents for a combat -- must check it, not all permanents.